includes/class-sensei-quiz.php

```diff
@@ -1451,6 +1451,18 @@ public static function the_user_status_message( $quiz_id ) {
 
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Output escaped above.
 		echo $message;
+
+		// show incomplete questions info message if there are incomplete questions and is preview.
+		if ( is_preview() ) {
+			$lesson_id          = Sensei()->quiz->get_lesson_id( $quiz_id );
+			$all_questions      = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ) );
+			$filtered_questions = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ), 'any', 'meta_value_num title', 'ASC', true );
+
+			if ( count( $all_questions ) !== count( $filtered_questions ) ) {
+				Sensei()->notices->add_notice( __( 'One or more questions in this lesson are incomplete. Incomplete questions are only displayed in preview mode and will not be displayed to students.', 'sensei-lms' ), 'info', 'incomplete-keys-messages' );
```

Comment:
As @dadish suggested, the message could be simplified a bit.

Reply:
Hey, I think we need to ask @donnapep about the message. I'm good with every solution

Reply:
Ah yes. I see the problem. There is actually no way to preview a quiz directly from the editor. There are 2 options that I can see:

1. The [original suggestion](https://github.com/Automattic/sensei/pull/4571#issuecomment-1004026536), which is to show the notice only to admins or the course teacher / lesson author.
2. Since quizzes can't be previewed directly, don't bother showing a notice at all. This would be consistent with the scenario we originally modelled this one after - draft lessons in a course. When viewing a course that has draft lessons in it, the admin / teacher only sees the published lessons (no draft ones).

Since I tend to value consistency and simplicity, I'm partial to #2, but what does everyone else think?

Reply:
I would vote for the option 2 as well.

My concern was that there is a difference in how it looks in WYSIWYG editor and the frontend. But I just checked and it looks like we're showing a notice in the editor's right sidebar that there are issues with the questions. Which I think is enough.